### PR TITLE
Fix diff-shared.sh and setup-repo.sh crash on --help

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Fix: diff-shared.sh and setup-repo.sh crash on --help by consume all parameters and then show the usage if requested

## Commits

- fix: diff-shared.sh and setup-repo.sh crash on --help by consume all parameters and then show the usage if requested

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [ ] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: scripts/bash/lib/_args.sh [hunk 1]
```diff
@@ -367,6 +368,9 @@ function get_table_format()
     return "$success"
 }
 
+declare -x usage_requested=""
+declare -x vm2_option="${VM2_REPOS:-}"
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply scripts/bash/lib/_args.sh]: removed

File: scripts/bash/lib/_args.sh [hunk 2]
```diff
@@ -409,6 +413,25 @@ function get_common_arg()
     return "$positive" # it was a common argument and was processed
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply scripts/bash/lib/_args.sh:412]: usage_requested assigned values. Thank you!

File: scripts/bash/repo-setup.sh [hunk 3]
```diff
@@ -47,7 +47,7 @@ declare -x use_https=false
 declare -x repo_owner=${ORGANIZATION:-$default_owner}
 declare -rx sot=$default_sot
 
-declare -x vm2_repos=""
+declare -x vm2_repos="${VM2_REPOS:-}"
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply scripts/bash/repo-setup.sh]: assigned default

File: scripts/bash/diff-shared.sh [hunk 4]
```diff
@@ -54,7 +54,7 @@ source "${script_dir}/diff-shared.usage.sh"
 #===============================
 # arguments:
 #===============================
-declare -x vm2_repos=""
+declare -x vm2_repos="${VM2_REPOS:-}"
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply scripts/bash/diff-shared.sh]: Assigned default
